### PR TITLE
consensus key rotation

### PR DESCRIPTION
**Pull request by zjma**
### Description

A step-by-step guide of how to rotate consensus key.

### Checklist

- Do all Lints pass?
  - [ ] Have you ran `pnpm spellcheck`?
  - [ ] Have you ran `pnpm fmt`?
  - [ ] Have you ran `pnpm lint`?

### CODE REVIEW

File: apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+Below is the step-by-step guide of performing a consensus key rotation with detailed examples.
+
+## New key generation
+
+Using the following [aptos CLI](https://aptos.dev/en/build/cli) command to generate a new consensus identity.
+In this example, identity files are saved to directory `/new/key/root`.
+```bash
+aptos genesis generate-keys --output-dir /new/key/root
+```
```
Comment (gregnazario) [inline apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx:21]:
```suggestion
```bash filename="Terminal"
aptos genesis generate-keys --output-dir /new/key/root
```
```

File: apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+Consensus key rotation is an action taken by a node operator to change the identity of a validator they control.
+It happens typically when the corresponding private key is (potentially) lost/leaked,
+or every X months as a common security practice.
+
+Below is the step-by-step guide of performing a consensus key rotation with detailed examples.
+
+## New key generation
+
+Using the following [aptos CLI](https://aptos.dev/en/build/cli) command to generate a new consensus identity.
```
Comment (gregnazario) [inline apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx:17]:
```suggestion
Using the following [Aptos CLI](https://aptos.dev/en/build/cli) command to generate a new consensus identity.
```

File: apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+title: "Rotate consensus key"
+---
+
+import { Callout, Steps } from 'nextra/components';
+
+# Rotate consensus key
+
+Consensus key rotation is an action taken by a node operator to change the identity of a validator they control.
+It happens typically when the corresponding private key is (potentially) lost/leaked,
+or every X months as a common security practice.
```
Comment (gregnazario) [inline apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx:11]:
```suggestion
or every few months as a common security practice.
```

File: apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        "last_update": 1731372563,
+        "value": "0x221f6fbfefa0a40b84c88fbb546a0884977dcc56719a96ed4e5d69b6a4ff58c8"
+    },
+    "consensus_acb7859468ca85cf9935e64ebb2b9b3fa8187de42d541acebaf732365a0131eaa994098f9d0d7e6b8ddea8ef11e16c55": {
+        "data": "GetResponse",
+        "last_update": 1731383387,
+        "value": "0x18d5098b3819d5fb0fc208b8bb0946b263f961f60716fc4d10d4b010fdb89a55"
+    },
+    // ...
+}
```
Comment (gregnazario) [inline apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx:130]: To match the comment by graphite, which I agree for syntax highlighting
```suggestion
```js
{
    // ...
    "consensus": {
        "data": "GetResponse",
        "last_update": 1731372563,
        "value": "0x221f6fbfefa0a40b84c88fbb546a0884977dcc56719a96ed4e5d69b6a4ff58c8"
    },
    "consensus_acb7859468ca85cf9935e64ebb2b9b3fa8187de42d541acebaf732365a0131eaa994098f9d0d7e6b8ddea8ef11e16c55": {
        "data": "GetResponse",
        "last_update": 1731383387,
        "value": "0x18d5098b3819d5fb0fc208b8bb0946b263f961f60716fc4d10d4b010fdb89a55"
    },
    // ...
}
```

File: apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+                # ...
+                identity_blob_path: /old/key/root/validator-identity.yaml
+                overriding_identity_paths:              # new!
+                - /new/key/root/validator-identity.yaml # new!
+```
+<Callout type="info">
+Don't delete `identity_blob_path`. It is a required field.
+
+Field `overriding_identity_paths` is optional, and allows multiple identities to be specified.
+
```
Comment (gregnazario) [inline apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx:43]: Oof this is messy, no wonder there's a guide

File: apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+Among all the identities, the node finds the one that appears on-chain validator set, and uses it as the consensus key.
```
Comment (gregnazario) [inline apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx:44]: This blurb is confusing.

Maybe...
```suggestion
* Field `identity_blob_path` is required.
* Field `overriding_identity_paths` is optional, and allows multiple identities to be specified for rotation purposes.
* The node will choose the identity that matches the on-chain state.
```

File: apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+or delete items from `overriding_identity_paths`,
+due to the reasons below.
+
+There is a separate **secure storage** (typically a file `secure_storage.json`) hosting all identities.
+The content of `identity_blob_path` and `overriding_identity_paths` are added to the secure storage on node start, if missing.
+However, keys are never deleted automatically from the storage.
+
+To completely wipe out an old key, you will have to manually make changes to the secure storage.
+See the end of the guide for more details.
+</Callout>
```
Comment (gregnazario) [inline apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx:58]: Why are there two callouts?

I would just remove the callouts altogether

**Review by gregnazario — COMMENTED** [review]
Can we get someone like @JoshLind or @michelle-aptos to read this and tell us if it makes sense?

**Review by zekun000 — APPROVED** [review]
lgtm, I assume you tested the whol flow e2e on some network

File: apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx (hunk 7 above)
Comment (zjma) [reply apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx]: deleted this callout, and only kept the first one.

File: apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx (hunk 6 above)
Comment (zjma) [reply apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx]: fixed

File: apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx (hunk 4 above)
Comment (zjma) [reply apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx]: fixec

File: apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx (hunk 3 above)
Comment (zjma) [reply apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx]: fixed

File: apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx (hunk 2 above)
Comment (zjma) [reply apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx]: fixed

File: apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx (hunk 1 above)
Comment (zjma) [reply apps/nextra/pages/en/network/nodes/validator-node/modify-nodes/rotate-consensus-key.mdx]: fixed

**Comment by zjma** [comment]
> lgtm, I assume you tested the whol flow e2e on some network

yes, on a local net :D

**Pull request merged**